src/Microsoft.AspNet.SignalR.Core/Transports/ForeverTransport.cs

```diff
@@ -298,16 +309,21 @@ private static Task PerformSend(object state)
 
             context.Transport.Context.Response.ContentType = JsonUtility.JsonMimeType;
 
-            context.Transport.JsonSerializer.Serialize(context.State, context.Transport.OutputWriter);
-            context.Transport.OutputWriter.Flush();
+            using (var writer = context.Transport.CreateMemoryPoolWriter(context.Transport.Pool))
+            {
+                context.Transport.JsonSerializer.Serialize(context.State, writer);
+                writer.Flush();
+
+                context.Transport.Context.Response.Write(writer.Buffer);
+            }
 
             return TaskAsyncHelper.Empty;
         }
 
-        private class ForeverTransportContext
+        private struct ForeverTransportContext
```

Comment:
Change this back to a class. I'm not sure why I made it a struct, seems like it gets boxed anyways.